app.go

```diff
@@ -181,6 +183,15 @@ func (a *App) Setup() {
 		if c.HelpName == "" {
 			c.HelpName = fmt.Sprintf("%s %s", a.HelpName, c.Name)
 		}
+
+		fc := newFlagCategories()
```

Comment:
Move this out into a separate function, with a []Flag as argument

Reply:
Excellent 👍🏼